spring-batch-infrastructure/src/main/java/org/springframework/batch/item/database/AbstractCursorItemReader.java

```diff
@@ -424,6 +444,9 @@ protected void initializeConnection() {
 			else {
 				this.con = dataSource.getConnection();
 			}
+			if (this.connectionAutoCommit != null && this.con.getAutoCommit() != this.connectionAutoCommit) {
+				this.con.setAutoCommit(this.connectionAutoCommit);
+			}
```

Comment:
If we allow this, we need to set it back to what it was once we're done with it.